ui/pages/routes/routes.component.tsx

```diff
@@ -470,14 +489,61 @@ export default function Routes() {
       dispatch(automaticallySwitchNetwork(networkToAutomaticallySwitchTo));
     }
 
-    prevPropsRef.current = { isUnlocked, totalUnapprovedConfirmationCount };
+    prevPropsRef.current = {
+      isUnlocked,
+      totalUnapprovedConfirmationCount,
+    };
   }, [
     networkToAutomaticallySwitchTo,
     isUnlocked,
     totalUnapprovedConfirmationCount,
     dispatch,
   ]);
 
+  // Show multichain intro modal when state 2 is enabled
+  useEffect(() => {
+    const prevMultichainProps = prevMultichainRef.current;
+
+    const isOnboardingRoute =
+      location.pathname.includes('/onboarding') ||
+      location.pathname.includes('/initialize');
+    const isMainWalletArea =
+      location.pathname === DEFAULT_ROUTE ||
+      (getEnvironmentType() === 'popup' && !isOnboardingRoute);
+
+    // Show modal when:
+    // 1. Feature flag transitions from disabled to enabled (upgrade case)
+    // 2. Feature flag is enabled and user sees main wallet area for first time (fresh install case)
+    const flagJustEnabled =
+      isMultichainAccountsState2Enabled &&
+      !prevMultichainProps.isMultichainAccountsState2Enabled;
+    const flagEnabledFirstTime =
+      isMultichainAccountsState2Enabled &&
+      isMainWalletArea &&
+      !hasShownMultichainIntroModal;
```

Comment:
```suggestion
    const flagEnabledFirstTime =
      isMultichainAccountsState2Enabled &&
      isMainWalletArea;
```

I think that last condition is duplicative to the condition below